pallets/anchors-v2/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The anchor already exists.
+		AnchorAlreadyExists,
+
+		/// The personal anchor already exists.
+		PersonalAnchorAlreadyExists,
+
+		/// The anchor was not found in storage.
+		AnchorNotFound,
+
+		/// The personal anchor was not found in storage.
+		PersonalAnchorNotFound,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Sets an anchor for a document ID and version.
+		#[pallet::weight(T::WeightInfo::set_anchor())]
+		#[pallet::call_index(0)]
+		pub fn set_anchor(
+			origin: OriginFor<T>,
+			document_id: T::DocumentId,
+			document_version: T::DocumentVersion,
+			hash: T::Hash,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			// Only one anchor should be stored for a particular document ID and version.
+			ensure!(
+				Anchors::<T>::iter_prefix_values((document_id, document_version)).count() == 0,
+				Error::<T>::AnchorAlreadyExists
+			);
+
+			ensure!(
+				PersonalAnchors::<T>::get((account_id.clone(), document_id, document_version))
+					.is_none(),
+				Error::<T>::PersonalAnchorAlreadyExists
+			);
+
+			let deposit = AnchorDeposit::<T>::get();
+
+			T::Currency::reserve(&account_id, deposit)?;
+
+			let anchor = AnchorOf::<T> {
+				account_id: account_id.clone(),
+				document_id,
+				document_version,
+				hash,
+				deposit,
+			};
+
+			Anchors::<T>::insert(
+				(document_id, document_version),
+				account_id.clone(),
+				anchor.clone(),
+			);
+
+			PersonalAnchors::<T>::insert(
+				(account_id.clone(), document_id, document_version),
+				anchor,
+			);
+
+			Self::deposit_event(Event::AnchorAdded {
+				account_id,
+				document_id,
+				document_version,
+				hash,
+				deposit,
+			});
+
+			Ok(())
+		}
+
+		/// Removes an anchor for a document ID and version.
+		#[pallet::weight(T::WeightInfo::remove_anchor())]
+		#[pallet::call_index(1)]
+		pub fn remove_anchor(
+			origin: OriginFor<T>,
+			document_id: T::DocumentId,
+			document_version: T::DocumentVersion,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			ensure!(
+				PersonalAnchors::<T>::get((account_id.clone(), document_id, document_version))
+					.is_some(),
+				Error::<T>::PersonalAnchorNotFound
+			);
+
+			let anchor = Anchors::<T>::get((document_id, document_version), account_id.clone())
+				.ok_or(Error::<T>::AnchorNotFound)?;
+
+			T::Currency::unreserve(&account_id, anchor.deposit);
+
+			Anchors::<T>::remove((document_id, document_version), account_id.clone());
+			PersonalAnchors::<T>::remove((account_id.clone(), document_id, document_version));
+
+			Self::deposit_event(Event::AnchorRemoved {
+				account_id,
+				document_id,
+				document_version,
+				hash: anchor.hash,
+				deposit: anchor.deposit,
+			});
+
+			Ok(())
+		}
+
+		/// Set a new anchor deposit.
+		#[pallet::weight(T::WeightInfo::set_deposit())]
+		#[pallet::call_index(2)]
+		pub fn set_deposit(origin: OriginFor<T>, new_deposit: T::Balance) -> DispatchResult {
```

Comment:
NIT: I would call it `set_deposit_value` or similar, if not it seems like this call makes the deposit when sets the value for future deposits.